pkg/health/health.go

```diff
@@ -28,24 +30,33 @@ type Result struct {
 	Output  string
 }
 
-func (h ConsulHealthChecker) Service(serviceID string) (map[string][]Result, error) {
-	entries, _, err := h.client.Health().Service(serviceID, "", false, nil)
+func (h ConsulHealthChecker) Service(serviceID string) (map[string][][]Result, error) {
+	catalogEntries, _, err := h.client.Health().Service(serviceID, "", false, nil)
 	if err != nil {
 		return nil, util.Errorf("/health/service failed for %q: %s", serviceID, err)
 	}
 
-	ret := make(map[string][]Result)
-	for _, entry := range entries {
-		ret[entry.Node.Node] = make([]Result, 0, len(entry.Checks))
+	ret := make(map[string][][]Result)
+	for _, entry := range catalogEntries {
+		// get entry in KV store for this service/node
+		kvEntry, err := h.consulStore.GetHealth(serviceID, entry.Node.Node)
+		if err != nil {
+			return nil, util.Errorf("/health/service failed for %q on node %s: %s", serviceID, entry.Node.Node, err)
+		}
+
+		res := make([]Result, 0, len(entry.Checks))
 		for _, check := range entry.Checks {
-			ret[entry.Node.Node] = append(ret[entry.Node.Node], consulCheckToResult(*check))
+			res = append(res, consulCheckToResult(*check))
 		}
+		kvRes := []Result{consulWatchToResult(kvEntry)}
+		// place worst result from catalog and result from KV store in map
+		ret[entry.Node.Node] = [][]Result{kvRes, res}
```

Comment:
How are clients expected to use this nested array? Could you explain why this change is necessary?